src/flashdeberta/ops/flash_attention.py

```diff
@@ -198,69 +282,85 @@ def get_fwd_config(B, H, M, N, D, causal, disentangled=False):
                     BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 32, 2, 4
                 else:
                     BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 128, 3, 8
-    elif capability == (8, 6):
+    elif capability[0] == 8:
         if not causal:
             if D <= 64:
                 BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 64, 3, 4
             else:
                 BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 32, 2, 4
         else:  # causal
             if D <= 64:
-                # For (8,6) devices we boost BLOCK_M for disentangled relative attention.
                 if disentangled:
                     BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 64, 3, 4
                 else:
                     BLOCK_M, BLOCK_N, num_stages, num_warps = 64, 64, 3, 4
             else:
                 BLOCK_M, BLOCK_N, num_stages, num_warps = 128, 32, 2, 4
     else:
-        BLOCK_M, BLOCK_N, num_stages, num_warps = 32, 32, 1, 4
+        BLOCK_M, BLOCK_N, num_stages, num_warps = 16, 16, 10, 4
+    
+    # Calculate shared memory usage with current config
+    has_pos = disentangled
+    ATT_SPAN = att_span if has_pos else 0 
+    
+    dtype = torch.float16
+
+    shared_mem_usage = calculate_shared_memory_usage(
+        BLOCK_M, BLOCK_N, D, num_stages, dtype, 
+        has_c2p=has_pos, has_p2c=has_pos, ATT_SPAN=ATT_SPAN
+    )
+    
+    # If shared memory usage exceeds available, adjust parameters
+    # We prioritize reducing num_stages first, then block sizes
+
+    while shared_mem_usage > max_shared_memory and (BLOCK_M > 16 or BLOCK_N > 16 or num_stages > 1):
+        # First try reducing num_stages
+        if num_stages > 1:
+            num_stages -= 1
+        # Then try reducing block sizes
+        elif BLOCK_M > 32 and BLOCK_N > 32:
+            BLOCK_M //= 2
+            BLOCK_N //= 2
+        elif BLOCK_M > 32:
+            BLOCK_M //= 2
+        elif BLOCK_N > 32:
+            BLOCK_N //= 2
+        elif BLOCK_M > 16:
+            BLOCK_M //= 2
+        elif BLOCK_N > 16:
+            BLOCK_N //= 2
+        
+        # Recalculate with new parameters
+        shared_mem_usage = calculate_shared_memory_usage(
+            BLOCK_M, BLOCK_N, D, num_stages, dtype, 
+            has_c2p=has_pos, has_p2c=has_pos, ATT_SPAN=ATT_SPAN
+        )
+    print(f"INFO: Forward config is {BLOCK_M}, {BLOCK_N}, {num_stages}, {num_warps} for BLOCK_M, BLOCK_N stages and warps, respectfully.")
+    print("INFO: If you want to change it, feel free to check ops/flash_attention")
```

Comment:
You probably want to use something else than print. Here it prints multiple time for every forward pass. Maybe a `warnings` so it print only once, which is quite standard in the wider `transformers` space?